merlin/models/tf/blocks/core/transformations.py

```diff
@@ -113,6 +115,142 @@ def get_config(self):
+        # Fix output shape and downrank to match input rank.
+        if rank == 2:
+            # tf.sparse.cross_hashed output shape will always be None on the last
+            # dimension. Given our input shape restrictions, we want to force shape 1
+            # instead.
+            output = tf.reshape(output, [-1, 1])
+        elif rank == 1:
+            output = tf.reshape(output, [-1])
+        elif rank == 0:
+            output = tf.reshape(output, [])
+
+        # Encode outputs.
+        outputs = {}
+        outputs[output_name] = preprocessing_utils.encode_categorical_inputs(
+            output,
+            output_mode=self.output_mode,
+            depth=self.num_bins,
+            sparse=self.sparse,
+        )
+        return outputs
+
+    def compute_output_shape(self, input_shapes):
+        self._check_at_least_two_inputs(input_shapes)
+        return preprocessing_utils.compute_shape_for_encode_categorical(input_shapes[0])
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "num_bins": self.num_bins,
+                "output_mode": self.output_mode,
+                "sparse": self.sparse,
+            }
+        )
+        if self.schema:
+            config["schema"] = schema_utils.schema_to_tensorflow_metadata_json(self.schema)
+        return config
+
+    def _check_at_least_two_inputs(self):
+        if len(self.schema) < 2:
+            raise ValueError(
+                "`HashedCrossing` should be called on at least two features. "
+                f"Received: {len(self.schema)} schemas"
+            )
+        for name, column_schema in self.schema.column_schemas.items():
+            if Tags.CATEGORICAL not in column_schema.tags:
+                warnings.warn(
+                    f"Please make sure input features to be categorical, detect {name} "
+                    "has no categorical tag"
+                )
+
+    def _check_input_shape_and_type(self, inputs: TabularData) -> TabularData:
+        inputs_tensors = []
+        for name in self.schema.column_names:
+            inputs_tensors.append(inputs[name])
+        first_shape = inputs_tensors[0].shape.as_list()
+        rank = len(first_shape)
+        if rank > 2 or (rank == 2 and first_shape[-1] != 1):
+            raise ValueError(
+                "All `HashedCrossing` inputs should have shape `[]`, `[batch_size]` "
+                f"or `[batch_size, 1]`. Received: inputs shape={first_shape}"
+            )
+        if not all(x.shape.as_list() == first_shape for x in inputs_tensors):
+            raise ValueError(
+                "All `HashedCrossing` inputs should have equal shape. "
+                f"Received: inputs={inputs_tensors}"
+            )
+        # TODO: Consider transfer tensors to dense tensors, do not require users do it by themself?
+        if any(isinstance(x, (tf.RaggedTensor, tf.SparseTensor)) for x in inputs_tensors):
+            raise ValueError(
```

Comment:
Do we support call another transformation block inside this block? Or we should make `to_dense` as `pre` of a TabularBlock? And do we support it now?